test/AutoDiff/differentiable_attr_type_checking.swift

```diff
@@ -29,16 +29,35 @@ func no_jvp_or_vjp(_ x: Float) -> Float {
   return x * x
 }
 
+// Test duplicated `@differentiable` attributes.
+
 @differentiable // expected-error {{duplicate '@differentiable' attribute}}
-@differentiable
+@differentiable // expected-note {{other attribute declared here}}
 func dupe_attributes(arg: Float) -> Float { return arg }
 
-@differentiable(wrt: arg1) // expected-error {{duplicate '@differentiable' attribute}}
 @differentiable(wrt: arg1)
 @differentiable(wrt: arg2) // expected-error {{duplicate '@differentiable' attribute}}
-@differentiable(wrt: arg2)
+@differentiable(wrt: arg2) // expected-note {{other attribute declared here}}
 func dupe_attributes(arg1: Float, arg2: Float) -> Float { return arg1 }
 
+struct ComputedPropertyDupeAttributes<T : Differentiable> : Differentiable {
+  var value: T
+
+  @differentiable // expected-error {{duplicate '@differentiable' attribute}}
+  var computed1: T {
+    @differentiable // expected-note {{other attribute declared here}}
+    get { value }
+    set { value = newValue }
+  }
+
+  @differentiable(where T == Float) // expected-error {{duplicate '@differentiable' attribute}}
```

Comment:
This should in theory be valid because generic requirements are different. I remember there are more issues in later phases (e.g. generic requirements not mangled in symbols, etc). If that's why this diagnostic is here, could you please add a FIXME?

Reply:
[TF-482](https://bugs.swift.org/browse/TF-482) tracks uniquing `@differentiable` attributes based on generic requirements.
Added TODO comments in https://github.com/apple/swift/pull/25293/commits/6488369c033c263abecb5fa0da22ee7c52c413ff.